src/metabase/driver/postgres.clj

```diff
@@ -296,6 +300,57 @@
   [driver [_ arg]]
   (sql.qp/->honeysql driver [:percentile arg 0.5]))
 
+(defmethod sql.qp/->honeysql [:postgres :datetimediff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)
+        helper
+        (fn helper [x y unit]
+          (case unit
+            (:year :day)
+            (hx/cast
+             :integer
```

Comment:
nit
```suggestion
            (hx/->integer
```

the `hx/->integer` will not cast if the function is already an `integer`. I guess it's not needed in this case but it'd be better if we keep using them.

Also, it saves 1 line of code.